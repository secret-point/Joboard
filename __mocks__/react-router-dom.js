// Mock navigate and useParams when needed
// const navigate = jest.fn();
// const useParams = jest.fn();
const useLocation = jest.fn();

module.exports = {
  ...jest.requireActual("react-router-dom"),
  // useNavigate: () => navigate,
  // useParams,
  useLocation,
};
