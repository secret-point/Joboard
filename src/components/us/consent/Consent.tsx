import React, { useEffect } from "react";
import { connect } from "react-redux";
import { Col } from "@amzn/stencil-react-components/layout";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { FlyoutContent, WithFlyout } from "@amzn/stencil-react-components/flyout";
import { Text } from "@amzn/stencil-react-components/text";
import InnerHTML from "dangerously-set-html-content";
import { getLocale, routeToAppPageWithPath } from "../../../utils/helper";
import { PAGE_ROUTES } from "../../pageRoutes";
import { getPageNameFromPath, parseQueryParamsArrayToSingleItem } from "../../../helpers/utils";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import queryString from "query-string";
import { Application } from "../../../utils/types/common";
import { useLocation } from "react-router";
import { JobState } from "../../../reducers/job.reducer";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import {
    boundCreateApplicationAndSkipScheduleDS,
    boundCreateApplicationDS
} from "../../../actions/ApplicationActions/boundApplicationActions";
import { CreateApplicationAndSkipScheduleRequestDS, CreateApplicationRequestDS } from "../../../utils/apiTypes";
import { uiState } from "../../../reducers/ui.reducer";
import { QUERY_PARAMETER_NAME } from "../../../utils/enums/common";
import { translate as t } from "../../../utils/translator";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import { ScheduleState } from "../../../reducers/schedule.reducer";

interface MapStateToProps {
    job: JobState;
    schedule: ScheduleState;
    ui: uiState;
}

interface RenderFlyoutFunctionParams {
    close: () => void;
}

const ConsentPage = (props: MapStateToProps) => {
    const { job, ui, schedule } = props;
    const isLoading = ui.isLoading;
    const { search, pathname } = useLocation();
    const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
    const jobId = queryParams.jobId;
    const jobDetail = job.results;
    const scheduleId = queryParams.scheduleId;
    const scheduleDetail = schedule.results.scheduleDetail;
    const pageName = getPageNameFromPath(pathname);
    const qualificationCriteria = jobDetail?.qualificationCriteria || [];
    const { JOB_OPPORTUNITIES } = PAGE_ROUTES;

    const isCreateButtonDisabled = scheduleId
        ? jobDetail && scheduleDetail && !isLoading ? false : true
        : jobDetail && !isLoading ? false : true;

    // Don't refetch data if id is not changing
    useEffect(() => {
        jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() })
    }, [jobDetail, jobId]);

    useEffect(() => {
        scheduleId && boundGetScheduleDetail({
            locale: getLocale(),
            scheduleId: scheduleId
        })
    }, [scheduleId]);

    useEffect(() => {
        jobDetail && addMetricForPageLoad(pageName)
    }, [jobDetail, pageName])

    const renderFlyout = ({ close }: RenderFlyoutFunctionParams) => (
        <FlyoutContent
            titleText={t("BB-ConsentPage-flyout-user-data-policy-title-text", "User Data Policy")}
            onCloseButtonClick={close}
        >
            <Col gridGap={15}>
                <Text>
                    <InnerHTML html={t("BB-ConsentPage-flyout-user-data-policy-paragraph1-text", "Amazon is committed to a diverse and inclusive workplace. Amazon is an equal opportunity employer and does not discriminate on the basis of race, national origin, gender, gender identity, sexual orientation, protected veteran status, disability, age, or other legally protected status. For individuals with disabilities who would like to request an accommodation, please visit <a href='https://www.amazon.jobs/en/disability/us' target='_blank' rel='noopener noreferrer'>https://www.amazon.jobs/en/disability/us</a>")} />
                </Text>
                <Text fontSize="T100">
                    <InnerHTML html={t("BB-ConsentPage-flyout-user-data-policy-paragraph2-text", "Amazon takes your personal data protection seriously and will only process your personal data in accordance with the law. In particular, we do not share your personal data with any third party without your prior consent, unless compelled by law to do so. Unless you express otherwise, the hiring Amazon company will store your personal data in the electronic database maintained by Amazon.com in the USA or one of its affiliates. For full information on how Amazon stores and processes your personal data, please click here <a href='http://www.amazon.jobs/' target='_blank' rel='noopener noreferrer'>http://www.amazon.jobs/</a>")} />
                </Text>
                <Text fontSize="T100">
                    {t("BB-ConsentPage-flyout-user-data-policy-paragraph3-text", "In addition to the role you applied for today, we may have other positions that require the same or similar job functions. These roles may be full-time, reduced-time, part-time, seasonal, and/or temporary positions that are available at this location or other nearby Amazon locations. You are under no obligation to accept any role with Amazon or our partners, however, by completing your application you agree to be considered for same level, hourly roles within a 30 mile radius of your application site. This includes sharing your applicant information with our partners (staffing agencies) that hire for both our regular and temporary associate needs.")}
                </Text>
                <Text fontSize="T100">
                    {t("BB-ConsentPage-flyout-user-data-policy-by-applying-text", "By applying, you:")}
                </Text>
                <ul className="ul-list">
                    <li>
                        <Text fontSize="T100">
                            {t("BB-ConsentPage-flyout-user-data-policy-by-applying-acknowledging-item1-text", "are acknowledging that you have read the job description for the position you are applying for and that you understand the basic requirements needed to perform the job;")}
                        </Text>
                    </li>
                    <li>
                        <Text fontSize="T100">
                            {t("BB-ConsentPage-flyout-user-data-policy-by-applying-acknowledging-item2-text", "consent to the processing, analyzing and assessment of your personal data by Amazon, salesforce.com or any other third party for the purposes of your application and for any other legitimate purposes of Amazon. For the avoidance of doubt, the \"processing\" of your personal data will include but not restricted to collecting, receiving, recording, organizing, collating, storing, updating, altering, using, disseminating, distributing, merging, linking, blocking, degrading, erasing or destroying of your personal data;")}
                        </Text>
                    </li>
                    <li>
                        <Text fontSize="T100">
                            {t("BB-ConsentPage-flyout-user-data-policy-by-applying-acknowledging-item3-text", "consent to Amazon retaining your personal data after the application process, in order to assist it with the effective monitoring of its job application processes and to your personal data being stored in an electronic database in the USA.")}
                        </Text>
                    </li>
                </ul>
            </Col>
        </FlyoutContent>
    )

    return (
        <Col gridGap="m" padding="n">
            <h1>
                {t("BB-ConsentPage-qualification-criteria-header-text", "By applying, you confirm that:")}
            </h1>
            <ul>
                {
                    qualificationCriteria.map(item => (
                        <li key={item}>{item}</li>
                    ))
                }
            </ul>
            <dl>
                <Text textAlign="center" color="gray" fontSize="0.8em">
                    {t("BB-ConsentPage-data-policy-header-text", "By applying, you read and agree to the")}
                </Text>
                <WithFlyout renderFlyout={renderFlyout}>
                    {({ open }) => (
                        <Button
                            variant={ButtonVariant.Tertiary}
                            style={{
                                margin: "0.5em 0",
                                width: "100%"
                            }}
                            onClick={() => open()}
                        >
                            {t("BB-ConsentPage-user-data-Policy-button", "User Data Policy")}
                        </Button>
                    )}
                </WithFlyout>
                <Button
                    variant={ButtonVariant.Primary}
                    style={{ width: "100%" }}
                    disabled={isCreateButtonDisabled}
                    onClick={() => {
                        if (scheduleId) {
                            const payload: CreateApplicationAndSkipScheduleRequestDS = {
                                jobId,
                                scheduleId,
                                dspEnabled: job.results?.dspEnabled,
                            }
                            boundCreateApplicationAndSkipScheduleDS(payload);
                        } else {
                            const payload: CreateApplicationRequestDS = {
                                jobId,
                                dspEnabled: job.results?.dspEnabled,
                            }
                            boundCreateApplicationDS(payload, (application: Application) => routeToAppPageWithPath(JOB_OPPORTUNITIES, [{ paramName: QUERY_PARAMETER_NAME.APPLICATION_ID, paramValue: application.applicationId }]));
                        }
                    }}
                >
                    {t("BB-ConsentPage-create-application-button", "Create Application")}
                </Button>
            </dl>
        </Col>
    );
};

const mapStateToProps = (state: MapStateToProps) => {
    return state;
};

export default connect(mapStateToProps)(ConsentPage);
