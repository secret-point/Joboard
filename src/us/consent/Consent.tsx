import React from "react";
import { connect } from "react-redux";
import { Col, View } from "@amzn/stencil-react-components/layout";
import { PageContainer } from "@amzn/stencil-react-components/page";
import { MainWithSkipLink } from "@amzn/stencil-react-components/a11y";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { FlyoutContent, WithFlyout } from "@amzn/stencil-react-components/flyout";
import { Text } from "@amzn/stencil-react-components/text";
import { parseQueryParamsArrayToSingleItem } from "../../helpers/utils";

interface MapStateToProps {

}

interface RenderFlyoutFunctionParams {
    close: () => void;
}

const ConsentPage = (props: MapStateToProps) => {
    const renderFlyout = ({ close }: RenderFlyoutFunctionParams) => (
        <FlyoutContent
            titleText="User Data Policy"
            onCloseButtonClick={close}
            buttons={[
                <Button onClick={close} variant={ButtonVariant.Primary}>
                Done
                </Button>
            ]}
        >
            <h2>User Data Policy</h2>
        </FlyoutContent>
    )

    return (
        <Col gridGap="m" padding="n">
            <PageContainer
            data-testid="layout"
            paddingTop="0"
            paddingBottom="0"
            paddingHorizontal="0"
            >
            <MainWithSkipLink>
            <h1>By applying, you confirm that:</h1>
            <ul>
                <li>You are at least 18 years old.</li>
                <li>You have at least a high school diploma or equivalent.</li>
                <li>You are willing to submit a pre-employment drug test.</li>
            </ul>
            <dl>
                <Text textAlign="center" color="gray" fontSize="0.8em">By applying, you read and agree to the</Text>
                <WithFlyout renderFlyout={renderFlyout}>
                    {({ open }) => (
                        <Button 
                            variant={ButtonVariant.Tertiary} 
                            style={{
                                margin:"0.5em 0",
                                width:"100%"
                            }} 
                            onClick={() => open()}>User Data Policy</Button>
                    )}
                </WithFlyout>
                <Button variant={ButtonVariant.Primary} style={{width:"100%"}} onClick={()=>{}}>Create Application</Button>
            </dl>
            </MainWithSkipLink>
        </PageContainer>
        </Col>
    );
};

const mapStateToProps = (state: MapStateToProps) => {
    return state;
};

export default connect(mapStateToProps)(ConsentPage);
