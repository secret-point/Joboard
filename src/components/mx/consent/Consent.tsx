import React, { useEffect } from "react";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { FlyoutContent, WithFlyout } from "@amzn/stencil-react-components/flyout";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import InnerHTML from "dangerously-set-html-content";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import {
    boundCreateApplicationAndSkipScheduleDS,
    boundCreateApplicationDS
} from "../../../actions/ApplicationActions/boundApplicationActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import { boundResetBannerMessage } from "../../../actions/UiActions/boundUi";
import {
    getPageNameFromPath,
    parseQueryParamsArrayToSingleItem,
    resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { JobState } from "../../../reducers/job.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { uiState } from "../../../reducers/ui.reducer";
import { CreateApplicationAndSkipScheduleRequestDS, CreateApplicationRequestDS } from "../../../utils/apiTypes";
import { QUERY_PARAMETER_NAME } from "../../../utils/enums/common";
import { getLocale, routeToAppPageWithPath } from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import { Application } from "../../../utils/types/common";
import DebouncedButton from "../../common/DebouncedButton";
import { PAGE_ROUTES } from "../../pageRoutes";

interface MapStateToProps {
    job: JobState;
    schedule: ScheduleState;
    ui: uiState;
}

interface RenderFlyoutFunctionParams {
    close: () => void;
}

export const Consent = (props: MapStateToProps) => {
    const { job, ui, schedule } = props;
    const { isLoading } = ui;
    const { search, pathname } = useLocation();
    const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
    const { jobId } = queryParams;
    const jobDetail = job.results;
    const { scheduleId } = queryParams;
    const { scheduleDetail } = schedule.results;
    const pageName = getPageNameFromPath(pathname);
    const qualificationCriteria = jobDetail?.qualificationCriteria || [];
    const { JOB_OPPORTUNITIES } = PAGE_ROUTES;

    const isCreateButtonDisabled = scheduleId
        ? !(jobDetail && scheduleDetail && !isLoading)
        : !(jobDetail && !isLoading);

    // Don't refetch data if id is not changing
    useEffect(() => {
        jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() })
    }, [jobId]);

    useEffect(() => {
        scheduleId && boundGetScheduleDetail({
            locale: getLocale(),
            scheduleId: scheduleId
        })
    }, [scheduleId]);

    useEffect(() => {
        if(jobDetail && ((scheduleId && scheduleDetail) || (!scheduleId))) {
            addMetricForPageLoad(pageName);
        }

    }, [jobDetail, scheduleDetail])

    useEffect(() => {
        return () => {
            //reset this so as it can emit new pageload event after being unmounted.
            resetIsPageMetricsUpdated(pageName);
        }
    },[])

    const renderFlyout = ({ close }: RenderFlyoutFunctionParams) => (
        <FlyoutContent
            titleText={t("BB-ConsentPage-flyout-user-data-policy-title-text", "User Data Policy")}
            onCloseButtonClick={close}
        >
            <Col gridGap={15}>
                <Text>
                    <InnerHTML html={t("BB-ConsentPage-flyout-user-data-policy-paragraph1-text", "Amazon is committed to a diverse and inclusive workplace. Amazon is an equal opportunity employer and does not discriminate on the basis of race, national origin, gender, gender identity, sexual orientation, protected veteran status, disability, age, or other legally protected status. For individuals with disabilities who would like to request an accommodation, please visit <a href='https://www.amazon.jobs/en/disability/us' target='_blank' rel='noopener noreferrer'>https://www.amazon.jobs/en/disability/us</a>")} />
                </Text>
                <Text fontSize="T100">
                    <InnerHTML html={t("BB-ConsentPage-flyout-user-data-policy-paragraph2-text", "Amazon takes your personal data protection seriously and will only process your personal data in accordance with the law. In particular, we do not share your personal data with any third party without your prior consent, unless compelled by law to do so. Unless you express otherwise, the hiring Amazon company will store your personal data in the electronic database maintained by Amazon.com in the USA or one of its affiliates. For full information on how Amazon stores and processes your personal data, please click here <a href='http://www.amazon.jobs/' target='_blank' rel='noopener noreferrer'>http://www.amazon.jobs/</a>")} />
                </Text>
                <Text fontSize="T100">
                    {t("BB-ConsentPage-flyout-user-data-policy-paragraph3-text", "In addition to the role you applied for today, we may have other positions that require the same or similar job functions. These roles may be full-time, reduced-time, part-time, seasonal, and/or temporary positions that are available at this location or other nearby Amazon locations. You are under no obligation to accept any role with Amazon or our partners, however, by completing your application you agree to be considered for same level, hourly roles within a 30 mile radius of your application site. This includes sharing your applicant information with our partners (staffing agencies) that hire for both our regular and temporary associate needs.")}
                </Text>
                <Text fontSize="T100">
                    {t("BB-ConsentPage-flyout-user-data-policy-by-applying-text", "By applying, you:")}
                </Text>
                <ul className="ul-list">
                    <li>
                        <Text fontSize="T100">
                            {t("BB-ConsentPage-flyout-user-data-policy-applying-acknowledging-item1-text", "are acknowledging that you have read the job description for the position you are applying for and that you understand the basic requirements needed to perform the job;")}
                        </Text>
                    </li>
                    <li>
                        <Text fontSize="T100">
                            {t("BB-ConsentPage-flyout-user-data-policy-applying-acknowledging-item2-text", "consent to the processing, analyzing and assessment of your personal data by Amazon, salesforce.com or any other third party for the purposes of your application and for any other legitimate purposes of Amazon. For the avoidance of doubt, the \"processing\" of your personal data will include but not restricted to collecting, receiving, recording, organizing, collating, storing, updating, altering, using, disseminating, distributing, merging, linking, blocking, degrading, erasing or destroying of your personal data;")}
                        </Text>
                    </li>
                    <li>
                        <Text fontSize="T100">
                            {t("BB-ConsentPage-flyout-user-data-policy-applying-acknowledging-item3-text", "consent to Amazon retaining your personal data after the application process, in order to assist it with the effective monitoring of its job application processes and to your personal data being stored in an electronic database in the USA.")}
                        </Text>
                    </li>
                </ul>
            </Col>
        </FlyoutContent>
    )

    return (
        <Col gridGap="m" padding="n">
            <h1>
                {t("BB-ConsentPage-qualification-criteria-header-text", "By applying, you confirm that:")}
            </h1>
            <ul>
                {
                    qualificationCriteria.map(item => (
                        <li key={item}>{item}</li>
                    ))
                }
            </ul>
            <Text>
                {t("BB-ConsentPage-whatsapp-consent-text", "I authorize Amazon to contact me via phone, text message or WhatsApp to provide information about my application, selection and hiring process.")}
            </Text>
            <dl>
                <Text textAlign="center" color="gray" fontSize="0.8em">
                    {t("BB-ConsentPage-data-policy-header-text", "By applying, you read and agree to the")}
                </Text>
                <WithFlyout renderFlyout={renderFlyout}>
                    {({ open }) => (
                        <Button
                            variant={ButtonVariant.Tertiary}
                            style={{
                                margin: "0.5em 0",
                                width: "100%"
                            }}
                            onClick={() => open()}
                        >
                            {t("BB-ConsentPage-user-data-Policy-button", "User Data Policy")}
                        </Button>
                    )}
                </WithFlyout>
                <DebouncedButton
                    variant={ButtonVariant.Primary}
                    style={{ width: "100%" }}
                    disabled={isCreateButtonDisabled}
                    onClick={() => {
                        boundResetBannerMessage();
                        if (scheduleId) {
                            const payload: CreateApplicationAndSkipScheduleRequestDS = {
                                jobId,
                                dspEnabled: job.results?.dspEnabled,
                            }
                            boundCreateApplicationAndSkipScheduleDS(payload);
                        } else {
                            const payload: CreateApplicationRequestDS = {
                                jobId,
                                dspEnabled: job.results?.dspEnabled,
                            }
                            boundCreateApplicationDS(payload, (application: Application) => routeToAppPageWithPath(JOB_OPPORTUNITIES, [{ paramName: QUERY_PARAMETER_NAME.APPLICATION_ID, paramValue: application.applicationId }]));
                        }
                    }}
                >
                    {t("BB-ConsentPage-create-application-button", "Create Application")}
                </DebouncedButton>
            </dl>
        </Col>
    );
};

const mapStateToProps = (state: MapStateToProps) => {
    return state;
};

export default connect(mapStateToProps)(Consent);
