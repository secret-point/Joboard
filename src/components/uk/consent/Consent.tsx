import React, { useEffect } from "react";
import { Button, ButtonVariant } from "@amzn/stencil-react-components/button";
import { FlyoutContent, WithFlyout } from "@amzn/stencil-react-components/flyout";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { H3, Text } from "@amzn/stencil-react-components/text";
import queryString from "query-string";
import { connect } from "react-redux";
import { useLocation } from "react-router-dom";
import { addMetricForPageLoad } from "../../../actions/AdobeActions/adobeActions";
import {
  boundCreateApplicationAndSkipScheduleDS,
  boundCreateApplicationDS
} from "../../../actions/ApplicationActions/boundApplicationActions";
import { boundGetJobDetail } from "../../../actions/JobActions/boundJobDetailActions";
import { boundGetScheduleDetail } from "../../../actions/ScheduleActions/boundScheduleActions";
import { boundResetBannerMessage } from "../../../actions/UiActions/boundUi";
import {
  getPageNameFromPath,
  parseQueryParamsArrayToSingleItem,
  resetIsPageMetricsUpdated
} from "../../../helpers/utils";
import { JobState } from "../../../reducers/job.reducer";
import { ScheduleState } from "../../../reducers/schedule.reducer";
import { uiState } from "../../../reducers/ui.reducer";
import { CreateApplicationAndSkipScheduleRequestDS, CreateApplicationRequestDS } from "../../../utils/apiTypes";
import { QUERY_PARAMETER_NAME } from "../../../utils/enums/common";
import { getLocale, routeToAppPageWithPath } from "../../../utils/helper";
import { translate as t } from "../../../utils/translator";
import { Application } from "../../../utils/types/common";
import DebouncedButton from "../../common/DebouncedButton";
import { PAGE_ROUTES } from "../../pageRoutes";
import InnerHTML from "dangerously-set-html-content";
import { Link } from "@amzn/stencil-react-components/link";

interface MapStateToProps {
  job: JobState;
  schedule: ScheduleState;
  ui: uiState;
}

interface RenderFlyoutFunctionParams {
  close: () => void;
}

export const Consent = (props: MapStateToProps) => {
  const { job, schedule } = props;
  const { search, pathname } = useLocation();
  const queryParams = parseQueryParamsArrayToSingleItem(queryString.parse(search));
  const { jobId } = queryParams;
  const jobDetail = job.results;
  const { scheduleId } = queryParams;
  const { scheduleDetail } = schedule.results;
  const pageName = getPageNameFromPath(pathname);
  const { JOB_OPPORTUNITIES } = PAGE_ROUTES;

  // Don't refetch data if id is not changing
  useEffect(() => {
    jobId && jobId !== jobDetail?.jobId && boundGetJobDetail({ jobId: jobId, locale: getLocale() });
  }, [jobId]);

  useEffect(() => {
    scheduleId && boundGetScheduleDetail({
      locale: getLocale(),
      scheduleId: scheduleId
    });
  }, [scheduleId]);

  useEffect(() => {
    if (jobDetail && ((scheduleId && scheduleDetail) || (!scheduleId))) {
      addMetricForPageLoad(pageName);
    }

  }, [jobDetail, scheduleDetail]);

  useEffect(() => {
    return () => {
      // reset this so as it can emit new pageload event after being unmounted.
      resetIsPageMetricsUpdated(pageName);
    };
  }, []);

  const renderFlyout = ({ close }: RenderFlyoutFunctionParams) => (
    <FlyoutContent
      titleText={t("BB-Kondo-ConsentPage-flyout-user-data-policy-title-text", "EU/UK Candidate Privacy Notice")}
      onCloseButtonClick={close}
    >
      <Col gridGap={15}>
        <Text>
          {t("BB-ConsentPage-flyout-mx-user-data-policy-text", "I am aware that all information gathered may be transferred pursuant to the Federal Law on the Protection of Personal Data in the Possession of Private Parties and its regulations, guidelines and other applicable provisions on data privacy, as well as in terms of the Privacy Notice of the Company for Employees and/or Candidates.")}
        </Text>
      </Col>
    </FlyoutContent>
  );
  const renderRightToWorkFlyout = ({ close }: RenderFlyoutFunctionParams) => (
    <FlyoutContent
      titleText={t("BB-Kondo-ConsentPage-flyout-right-to-work-title-text", "Right To Work documentation")}
      onCloseButtonClick={close}
    >
      <Col gridGap={15}>
        <H3 fontWeight="bold">
          {t("BB-Kondo-ConsentPage-flyout-uk-right-to-work-content-title-text", "UK Right to Work Check")}
        </H3>
        <Text>
          {t("BB-Kondo-ConsentPage-flyout-uk-right-to-work-part1-text", "In order to begin your employment, Amazon must check your original document(s) proving your right to work in the UK. Right to Work document examples include, but are not limited to")}:
        </Text>
        <ul>
          <li>
            <Text>
              {t("BB-Kondo-ConsentPage-flyout-uk-right-to-work-document-item-passport-text", "British Passport")}
            </Text>
          </li>
          <li>
            <Text>
              {t("BB-Kondo-ConsentPage-flyout-uk-right-to-work-document-item-residence-card-text", "Permanent Residence Card")}
            </Text>
          </li>
          <li>
            <Text>
              {t("BB-Kondo-ConsentPage-flyout-uk-right-to-work-document-item-indefinite-leave-stamp-text", "Indefinite Leave to Remain stamps")}
            </Text>
          </li>
          <li>
            <Text>
              {t("BB-Kondo-ConsentPage-flyout-uk-right-to-work-document-item-biometric-residence-permit-text", "Biometric Residence Permit")}
            </Text>
          </li>
        </ul>
        <Text>
          <InnerHTML html={t("BB-Kondo-ConsentPage-flyout-uk-right-to-work-part2-text", "For more information and details of all acceptable Right to Work Documents, please see our <a href='https://glspub.s3-us-west-2.amazonaws.com/Work+Authorization/UK+Employee+Guide/index.html' target='_blank' rel='noopener noreferrer'>UK Employee Work Authorisation Guide</a>")} />
        </Text>
      </Col>
    </FlyoutContent>
  );

  return (
    <Col gridGap="S300" padding="n">
      <h1>
        {t("BB-ConsentPage-qualification-criteria-header-text", "By applying, you confirm that:")}
      </h1>
      <ul>
        <li>
          <Text fontSize="T200">
            {t("BB-Kondo-qualification-criteria-above18", "You are 18 years or above.")}
          </Text>
        </li>
        <li>
          <Text fontSize="T200">
            {t("BB-Kondo-qualificationCriteria-should-complete-all-steps", "You are the sole applicant and shall be personally completing each step unless you receive any assistance from Amazon to complete this application.")}
          </Text>
        </li>
        <li>
          <Text fontSize="T200">
            {t("BB-Kondo-qualification-criteria-right-to-work-part1", "You have")}
            <WithFlyout renderFlyout={renderRightToWorkFlyout}>
              {({ open }) => (
                <Link
                  onClick={() => open()}
                  margin={{
                    left: "S100"
                  }}
                >
                  {t("BB-Kondo-qualificationCriteria-right-to-work-part2", "the right to work documentation")}
                </Link>
              )}
            </WithFlyout>
          </Text>
        </li>
        <li>
          <Text fontSize="T200">
            {t("BB-Kondo-qualificationCriteria-medical-test", "You are willing to complete a pre-employment medical test, if applicable.")}
          </Text>
        </li>
        <li>
          <Text fontSize="T200">
            {t("BB-Kondo-qualificationCriteria-employment-bgc-consent", "You consent to a pre-employment background check.")}
          </Text>
        </li>
      </ul>
      <Col gridGap={8}>
        <Text textAlign="center" color="gray" fontSize="0.8em">
          {t("BB-ConsentPage-data-policy-header-text", "By applying, you read and agree to the")}
        </Text>
        <WithFlyout renderFlyout={renderFlyout}>
          {({ open }) => (
            <Button
              variant={ButtonVariant.Tertiary}
              style={{
                margin: "0.5em 0",
                width: "100%"
              }}
              onClick={() => open()}
            >
              {t("BB-Kondo-ConsentPage-flyout-user-data-policy-title-text", "EU/UK Candidate Privacy Notice")}
            </Button>
          )}
        </WithFlyout>
        <DebouncedButton
          variant={ButtonVariant.Primary}
          style={{ width: "100%" }}
          onClick={() => {
            boundResetBannerMessage();
            if (scheduleId) {
              const payload: CreateApplicationAndSkipScheduleRequestDS = {
                jobId,
                dspEnabled: job.results?.dspEnabled,
              };
              boundCreateApplicationAndSkipScheduleDS(payload);
            } else {
              const payload: CreateApplicationRequestDS = {
                jobId,
                // dspEnabled: job.results?.dspEnabled, // TODO will need to enabled this in future once dspEnabled is enabled in UK
              };
              boundCreateApplicationDS(payload, (application: Application) => routeToAppPageWithPath(JOB_OPPORTUNITIES, [{ paramName: QUERY_PARAMETER_NAME.APPLICATION_ID, paramValue: application.applicationId }]));
            }
          }}
        >
          {t("BB-kondo-ConsentPage-start-application-button", "Start Application")}
        </DebouncedButton>
        <Col
          className="consentDisabilityAccommodationText"
          gridGap={15}
          padding={{ top: "S400", bottom: "S400", left: "S300", right: "S300" }}
          margin={{ top: "S400" }}
        >
          <Text fontSize="T100">
            <InnerHTML html={t("BB-Kondo-ConsentPage-consent-disability-accommodation-text", "Our inclusive culture empowers Amazonians to deliver the best results for our customers. If you have a disability and need an adjustment during the application and hiring process, including support for the interview or onboarding process, please contact the Applicant-Candidate Accommodation Team (ACAT), Monday through Friday from 7:00 am GMT - 4:00 pm GMT . If calling directly from the United Kingdom, please dial +44 800 086 9884. If calling from Ireland, please dial +353 1800 851 489. You may also contact us if you might need an adjustment in your new role and would like to initiate a request prior to starting your Day 1.")} />
          </Text>
        </Col>
      </Col>
    </Col>
  );
};

const mapStateToProps = (state: MapStateToProps) => {
  return state;
};

export default connect(mapStateToProps)(Consent);
