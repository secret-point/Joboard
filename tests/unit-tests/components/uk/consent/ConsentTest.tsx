import React from "react";
import { shallow } from "enzyme";
import { useLocation } from "react-router-dom";
import { Consent } from "../../../../../src/components/uk/consent/Consent";
import { TestInitUiState, TEST_JOB_ID, TEST_JOB_STATE, TEST_SCHEDULE_ID, TEST_SCHEDULE_STATE } from "../../../../test-utils/test-data";

describe("Consent", () => {
  const mockLocation = {
    pathname: "/consent",
    search: `?jobId=${TEST_JOB_ID}&scheduleId=${TEST_SCHEDULE_ID}`,
    hash: "",
    state: null
  };
  const mockUseLocation = useLocation as jest.Mock;
  mockUseLocation.mockReturnValue(mockLocation);

  it("should match snapshot", () => {
    const shallowWrapper = shallow(
      <Consent
        job={TEST_JOB_STATE}
        schedule={TEST_SCHEDULE_STATE}
        ui={TestInitUiState}
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });
});
