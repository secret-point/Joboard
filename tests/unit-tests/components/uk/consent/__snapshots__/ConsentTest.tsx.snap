// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consent should match snapshot 1`] = `
<Col
  gridGap="S300"
  padding="n"
>
  <h1>
    By applying, you confirm that:
  </h1>
  <ul>
    <li>
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
      >
        You are 18 years or above.
      </Text>
    </li>
    <li>
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
      >
        You are the sole applicant and shall be personally completing each step unless you receive any assistance from Amazon to complete this application.
      </Text>
    </li>
    <li>
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
      >
        You have
        <WithFlyout
          flyoutPosition="trailing"
          renderFlyout={[Function]}
        >
          <Component />
        </WithFlyout>
      </Text>
    </li>
    <li>
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
      >
        You are willing to complete a pre-employment medical test, if applicable.
      </Text>
    </li>
    <li>
      <Text
        font="primary"
        fontSize="T200"
        hyphens="none"
      >
        You consent to a pre-employment background check.
      </Text>
    </li>
  </ul>
  <Col
    gridGap={8}
  >
    <Text
      color="gray"
      font="primary"
      fontSize="0.8em"
      hyphens="none"
      textAlign="center"
    >
      By applying, you read and agree to the
    </Text>
    <WithFlyout
      flyoutPosition="trailing"
      renderFlyout={[Function]}
    >
      <Component />
    </WithFlyout>
    <DebouncedButton
      onClick={[Function]}
      style={
        Object {
          "width": "100%",
        }
      }
      variant="primary"
    >
      Start Application
    </DebouncedButton>
    <Col
      className="consentDisabilityAccommodationText"
      gridGap={15}
      margin={
        Object {
          "top": "S400",
        }
      }
      padding={
        Object {
          "bottom": "S400",
          "left": "S300",
          "right": "S300",
          "top": "S400",
        }
      }
    >
      <Text
        font="primary"
        fontSize="T100"
        hyphens="none"
      >
        <DangerouslySetHtmlContent
          html="Our inclusive culture empowers Amazonians to deliver the best results for our customers. If you have a disability and need an adjustment during the application and hiring process, including support for the interview or onboarding process, please contact the Applicant-Candidate Accommodation Team (ACAT), Monday through Friday from 7:00 am GMT - 4:00 pm GMT . If calling directly from the United Kingdom, please dial +44 800 086 9884. If calling from Ireland, please dial +353 1800 851 489. You may also contact us if you might need an adjustment in your new role and would like to initiate a request prior to starting your Day 1."
        />
      </Text>
    </Col>
  </Col>
</Col>
`;
